cassandraprotocol/frame/decoder.go

```diff
@@ -19,19 +19,30 @@ func (c *Codec) Decode(source io.Reader) (*Frame, error) {
 	} else {
 		header := &Header{
 			Version:          rawHeader.Version,
-			StreamId:         int16(rawHeader.StreamId & 0xFFFF),
+			StreamId:         rawHeader.StreamId,
 			TracingRequested: body.TracingId != nil || rawHeader.Flags&cassandraprotocol.HeaderFlagTracing != 0,
 		}
 		return &Frame{Header: header, Body: body}, nil
 	}
 }
 
+// Decode decodes the entre frame, decompressing the body if needed.
```

Comment:
```suggestion
// DecodeRaw decodes the entire frame, decompressing the body if needed, and returns a RawFrame.
```
(We need to fix the typo in the docs of `Decode` as well)

Reply:
done